modules/src/ics03_connection/context_mock.rs

```diff
@@ -11,11 +11,12 @@ use crate::ics24_host::identifier::{ClientId, ConnectionId};
 use std::collections::HashMap;
 use tendermint::block::Height;
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug)]
 pub struct MockConnectionContext {
     chain_context: MockChainContext,
     client_context: MockClientContext,
     connections: HashMap<ConnectionId, ConnectionEnd>,
+    client_connections: HashMap<ClientId, ConnectionId>,
```

Comment:
I think this should be in the client context. We can discuss and change it in a different PR.